test/e2e/dnslookup_test.go

```diff
@@ -0,0 +1,285 @@
+		_, err = cmdRunWithOutput("wsl", "--unregister", wslDistroName)
+		require.NoErrorf(t, err, "Failed to unregister distro %v", wslDistroName)
+	}()
+
+	// It takes a long time to start a new distro,
+	// 20 sec is a long time but that's actually how long
+	// it takes to start a distro without any flakiness
+	timeout := time.Second * 20
+	tryInterval := time.Second * 2
+	err = confirm(func() bool {
+		// Run `wslpath` to see if the distribution is registered; this avoids
+		// parsing the output of `wsl --list` to avoid having to handle UTF-16.
+		out, err := cmdRunWithOutput("wsl", "--distribution", wslDistroName, "--exec", "/bin/wslpath", ".")
+		if err != nil {
+			return false
+		}
+		// We expect `wslpath` to output a single dot for the given command.
+		return strings.TrimSpace(out) == "."
+	}, tryInterval, timeout)
+	require.NoErrorf(t, err, "Failed to check if %v wsl distro is running", wslDistroName)
+
+	dnsInfs, err := helper.GetDNSInterfaces()
+	require.NoError(t, err, "Failed getting DNS addrs associated to interfaces")
+
+	// restore the system DNS servers to the original state
+	defer restoreSystemDNS(t, dnsInfs)
+
+	t.Log("Updating network interfaces with test DNS server address")
+	// Update the dns addrs to test server
+	updateSystemDNS(t, testSrvAddr, dnsInfs)
+
+	aRecords := testdns.LoadRecords("../testdata/test-300.csv")
+
+	tcpHandler := &testdns.Handler{
+		Truncate: false,
+		Arecords: aRecords,
+	}
+
+	udpHandler := &testdns.Handler{
+		Truncate: true,
+		Arecords: aRecords,
+	}
+
+	testServer := testdns.Server{
+		Addr:       testSrvAddr,
+		TCPPort:    dnsPort,
+		UDPPort:    dnsPort,
+		TCPHandler: tcpHandler,
+		UDPHandler: udpHandler,
+	}
+	t.Log("Starting test upstream DNS server")
+	go testServer.Run()
+
+	t.Logf("Starting host-resolver peer process in wsl [%v]", wslDistroName)
+	peerCmd := cmdExec(
+		tmpDir,
+		"wsl", "--user", "root",
+		"--distribution", wslDistroName,
+		"--exec", "./rancher-desktop-host-resolver", "vsock-peer")
+	err = peerCmd.Start()
+	require.NoError(t, err, "Starting host-resolver peer process faild")
+	defer func() {
+		_ = peerCmd.Process.Kill()
+	}()
+
+	t.Log("Starting host-resolver host process")
+	resolverExecPath := fmt.Sprintf("%v/rancher-desktop-host-resolver.exe", tmpDir)
+	hostCmd := cmdExec(
+		tmpDir,
+		resolverExecPath, "vsock-host",
+		"--upstream-servers", fmt.Sprintf("[%v]", testSrvAddr))
+	err = hostCmd.Start()
+	require.NoError(t, err, "Starting host-resolver host process faild")
+	defer func() {
+		_ = hostCmd.Process.Kill()
+	}()
+
+	t.Log("Building DNS hammer binary")
+	err = buildBinaries("../...", "linux", tmpDir)
+	require.NoError(t, err, "Failed building dnsHammer")
+
+	err = copyFile("../testdata/test-300.csv", fmt.Sprintf("%s/test-300.csv", tmpDir))
+	require.NoError(t, err, "Failed copying test data file")
+
+	t.Log("Confirming host-resolver peer process is up")
+	peerCmdTimeout := time.Second * 10
+	confirm(func() bool { //nolint:errcheck
```

Comment:
Why are we ignoring the error from `confirm`? At least do `_ = confirm(…` instead of suppressing the linter?